packages/subscription/src/chainable-subscribable.ts

```diff
@@ -0,0 +1,62 @@
+import { Operation } from 'effection';
+import { ChainableSubscription } from './chainable-subscription';
+import { DeepPartial } from './match';
+import { Subscribable } from './subscribable';
+import { SymbolSubscribable } from './symbol-subscribable';
+import { Subscription } from './subscription';
+
+export interface ChainableSubscribableMethods<T,TReturn> extends Subscribable<T,TReturn> {
+  filter(predicate: (value: T) => boolean): ChainableSubscribable<T, TReturn>;
+  match(reference: DeepPartial<T>): ChainableSubscribable<T,TReturn>;
+  map<R>(mapper: (value: T) => R): ChainableSubscribable<R, TReturn>;
+  first(): Operation<T | undefined>;
+  expect(): Operation<T>;
+  forEach(visit: (value: T) => Operation<void>): Operation<TReturn>;
+  next(): Operation<IteratorResult<T, TReturn>>;
+}
+
+export type ChainableSubscribable<T, TReturn> = Operation<ChainableSubscription<T, TReturn>> & ChainableSubscribableMethods<T, TReturn>;
+
+export function makeChainable<T, TReturn>(
+  operation: Operation<ChainableSubscription<T, TReturn>>
+): ChainableSubscribable<T, TReturn> {
+  return Object.assign(operation, {
+    *[SymbolSubscribable](): Operation<Subscription<T,TReturn>> {
+      return yield operation;
+    },
+
+    filter(predicate: (value: T) => boolean): ChainableSubscribable<T, TReturn> {
+      return makeChainable(function*() {
+        return yield ChainableSubscription.wrap<T, TReturn>(operation, (inner) => inner.filter(predicate));
+      });
+    },
+
+    match(reference: DeepPartial<T>): ChainableSubscribable<T,TReturn> {
+      return makeChainable(function*() {
+        return yield ChainableSubscription.wrap<T, TReturn>(operation, (inner) => inner.match(reference));
+      });
+    },
+
+    map<R>(mapper: (value: T) => R): ChainableSubscribable<R, TReturn> {
+      return makeChainable(function*() {
+        return yield ChainableSubscription.wrap<T, TReturn, R>(operation, (inner) => inner.map(mapper));
+      });
+    },
+
+    *first(): Operation<T | undefined> {
+      return yield (yield operation).first();
```

Comment:
Ladies and gentlement, the first double yield 🙌

Reply:
Hopefully this will allow us to avoid the double yield in other places 😈